external/fv3fit/fv3fit/_shared/xr_prediction.py

```diff
@@ -70,3 +89,89 @@ def predict_on_dataset(
         unstacked_dims=unstacked_dims,
     )
     return match_prediction_to_input_coords(X, return_ds)
+
+
+@io.register("dataset-predictor")
+class DatasetPredictor(Predictor):
+    """
+    Base model is an ArrayPredictor which takes in a sequence of
+    input arrays and outputs a sequence of output arrays.
+    Normalization is contained within the base ArrayPredictor.
+
+    Assumes wrapped model accepts [sample, feature] arrays.
+    """
+
+    _BASE_MODEL_NAME = "base_model"
+    _CONFIG_FILENAME = "config.yaml"
+
+    def __init__(
+        self,
+        input_variables: Iterable[Hashable],
+        output_variables: Iterable[Hashable],
+        model: ArrayPredictor,
+        unstacked_dims: Sequence[str],
+        n_halo: int = 0,
+    ):
+        """Initialize the predictor
+
+        Args:
+            input_variables: names of input variables
+            output_variables: names of output variables
+            model: base model to wrap. Must be in fv3fit.io registry
+            unstacked_dims: non-sample dimensions of model output
+            n_halo: number of halo points required in input data
+        """
+        super().__init__(input_variables, output_variables)
+        self.input_variables = input_variables
+        self.output_variables = output_variables
+        self.model = model
+        self._n_halo = n_halo
+        self._unstacked_dims = unstacked_dims
```

Comment:
Given that our use case is typically that unstacked_dims = ["z"], I'm lean towards making that the default value here.  Similarly with n_halo = 0.